src/OpenTelemetry.Api/Logs/LoggerProvider.cs

```diff
@@ -25,33 +29,53 @@ internal class LoggerProvider : BaseProvider
 {
     private static readonly NoopLogger NoopLogger = new();
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="LoggerProvider"/> class.
-    /// </summary>
-    protected LoggerProvider()
-    {
-    }
-
     /// <summary>
     /// Gets a logger.
     /// </summary>
     /// <returns><see cref="Logger"/> instance.</returns>
     public Logger GetLogger()
-        => this.GetLogger(new InstrumentationScope());
+        => this.GetLogger(name: null, version: null);
 
     /// <summary>
     /// Gets a logger with the given name.
     /// </summary>
     /// <param name="name">Optional name identifying the instrumentation library.</param>
     /// <returns><see cref="Logger"/> instance.</returns>
-    public Logger GetLogger(string name)
-        => this.GetLogger(new InstrumentationScope(name));
+    public Logger GetLogger(string? name)
+        => this.GetLogger(name, version: null);
+
+    /// <summary>
+    /// Gets a logger with the given name and version.
+    /// </summary>
+    /// <param name="name">Optional name identifying the instrumentation library.</param>
+    /// <param name="version">Optional version of the instrumentation library.</param>
+    /// <returns><see cref="Logger"/> instance.</returns>
+    public Logger GetLogger(string? name, string? version)
+    {
+        if (!this.TryCreateLogger(name, out var logger))
+        {
+            return NoopLogger;
+        }
+
+        logger!.SetInstrumentationScope(version);
+
+        return logger;
+    }
 
     /// <summary>
-    /// Gets a logger with given instrumentation scope.
+    /// Try to create a logger with the given name.
     /// </summary>
-    /// <param name="instrumentationScope"><see cref="InstrumentationScope"/>.</param>
-    /// <returns><see cref="Logger"/>.</returns>
-    public virtual Logger GetLogger(InstrumentationScope instrumentationScope)
-        => NoopLogger;
+    /// <param name="name">Optional name identifying the instrumentation library.</param>
+    /// <param name="logger"><see cref="Logger"/>.</param>
+    /// <returns><see langword="true"/> if the logger was created.</returns>
+    protected virtual bool TryCreateLogger(
```

Comment:
This `TryCreateLogger` may look a little strange, but it accomplishes a number of things...

* There is only a single `virtual` method that must be implemented in SDK(s).
* We can add other `GetLogger` methods freely that add other things to the instrumentation scope (schema_url, attributes, etc.) maintained on `Logger` without implementations breaking.
* There is some stuff in the spec about repeat calls returning the same instances and mutating instances when config changes. We don't support that yet, but in theory separating "Get" from "Create" should allow the base class to implement that stuff if we ever need to.